AssemblerLib/LocalToGlobalIndexMap.h

```diff
@@ -65,6 +82,16 @@ class LocalToGlobalIndexMap
     LineIndex columnIndices(std::size_t const mesh_item_id) const;
 
 private:
+
+    /// Private constructor used by internally created local-to-global index
+    /// maps. The mesh_component_map is passed as argument instead of being
+    /// created by the constructor.
+    explicit LocalToGlobalIndexMap(
+        std::vector<MeshLib::MeshSubsets*> const& mesh_subsets,
+        std::vector<MeshLib::Element*> const& elements,
+        AssemblerLib::MeshComponentMap const& mesh_component_map,
+        AssemblerLib::ComponentOrder const order);
+
     template <typename ElementIterator>
```

Comment:
@TomFischer  Just think it is a little bit big function inside the class deflations. Besides I have to added some lines inside it ( #580). Looks like #580 will be merged after this PR. Otherwise, I would like you to take a look at #580.